### PR TITLE
refactor: facilitate make.sh build

### PR DESCRIPTION
**Pull request by u2takey**
Signed-off-by: u2takey <u2takey@gmail.com>

### Ⅰ. Describe what this PR did
facilitate make.sh build by checking install status before do install binaries

### CODE REVIEW

**Comment by HusterWan** [comment]
@u2takey Thanks a lot for your PR

@Letty5411 could your please review this PR?

File: hack/make.sh [hunk 1]
```diff
@@ -16,26 +20,64 @@ if [[ $SOURCEDIR != $DIR ]];then
...
+	if which dumb-init &>/dev/null; then
+		DUMB_INIT_VERSION=$(dumb-init -V 2>&1 | cut -d " " -f 2)
+	fi	
+}
+
+function install_containerd()
+{
+	echo "Try installing containerd"
+	get_containerd_version
+	if [ "$CONTAINERD_VERSION" = "v1.0.3" ]; then
```
Comment (Letty5411) [inline hack/make.sh]: `=` to be `==`
And I would prefer user `[[ ]]`, same comments for the following code.

File: hack/make.sh [hunk 2]
```diff
@@ -46,28 +88,61 @@ function install_pouch ()
...
-		./autogen.sh
-		autoreconf -vfi
-		./configure && make 
-		cp ./nsenter /usr/local/bin
-		cd $DIR/
+		if [[ "$NSENTER_VERSION" == "2.24.1" ]]; then
+			echo "Nsenter already installed."
+		else
+			echo "Download and install nsenter."
+			apt-get install libncurses5-dev libslang2-dev gettext zlib1g-dev libselinux1-dev debhelper lsb-release pkg-config po-debconf autoconf automake autopoint libtool
```
Comment (sunyuan3) [inline hack/make.sh]: apt-get install -y

File: hack/make.sh [hunk 3]
```diff
@@ -46,28 +88,61 @@ function install_pouch ()
...
+			wget https://www.kernel.org/pub/linux/utils/util-linux/v2.24/util-linux-2.24.1.tar.gz -P $TMP
```
Comment (sunyuan3) [inline hack/make.sh]: wget -q

File: hack/make.sh [hunk 4]
```diff
@@ -46,28 +88,61 @@ function install_pouch ()
...
 {
-    wget -O /usr/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.1/dumb-init_1.2.1_amd64 || return 1
-    chmod +x /usr/bin/dumb-init
+	echo "Try installing dumb-init"
+	get_dumb_init_version
+	if [[ "$DUMB_INIT_VERSION" == "v1.2.1" ]]; then
+		echo "Dumb-init already installed."
+	else
+		echo "Download and install dumb-init."
+		wget -O /tmp/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.1/dumb-init_1.2.1_amd64
```
Comment (sunyuan3) [inline hack/make.sh]: wget -q

**Comment by u2takey** [comment]
updated，ping @sunyuan3 @Letty5411 for review

**Comment by sunyuan3** [comment]
LGTM

**Pull request merged**